docker/snippets/rosdep.dockerfile

```diff
@@ -12,16 +12,29 @@ RUN apt-get update && \
 ARG ROS_DISTRO
 ARG ROS_WORKSPACE="./workspace"
 ARG ROSDEP_METAPACKAGE=""
+ARG ROSDEP_OS_NAME="ubuntu"
+ARG ROSDEP_OS_VERSION="jammy"
+ARG ROSDEP_SKIP_KEYS=""
 ARG ROS_INSTALL_PREFIX="/opt/ros/${ROS_DISTRO}/"
 COPY ${ROS_WORKSPACE}/src /tmp/workspace/src
 RUN . ${ROS_INSTALL_PREFIX}/setup.sh && \
     cd /tmp/workspace && \
     apt-get update && \
-    if [ -z "$ROSDEP_METAPACKAGE" ]; then \
+    if [[ -z "${ROSDEP_METAPACKAGE}" ]]; then \
         ROSDEP_FROM_PATHS="src"; \
     else \
-        ROSDEP_FROM_PATHS=$(colcon list --packages-up-to $ROSDEP_METAPACKAGE | awk '{print $2}' | tr '\n' ' '); \
+        ROSDEP_FROM_PATHS=$(colcon list --packages-up-to ${ROSDEP_METAPACKAGE} | awk '{print $2}' | tr '\n' ' '); \
     fi && \
-    rosdep install --from-paths $ROSDEP_FROM_PATHS --ignore-src -r -y && \
+    rosdep install --from-paths ${ROSDEP_FROM_PATHS} --ignore-src -r -y --skip-keys="${ROSDEP_SKIP_KEYS}" --os=${ROSDEP_OS_NAME}:${ROSDEP_OS_VERSION} --rosdistro=${ROS_DISTRO} && \
     rm -rf /tmp/workspace && \
     apt-get clean && apt-get autoremove -y && rm -rf /var/lib/apt/lists/*
+
+# Avoid setup.py and easy_install deprecation warnings caused by colcon and setuptools
+# https://github.com/colcon/colcon-core/issues/454
+ENV PYTHONWARNINGS=ignore:::setuptools.command.install,ignore:::setuptools.command.easy_install,ignore:::pkg_resources
```

Comment:
latest versions of setuptools would throw a lot of warnings with ros cause the latest versions of python no longer allow easy_install I believe. This line just makes it so those warnings go away. No longer to need to install an older version of setuptools to get rid of that warning. See the first change at the top.